sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -643,15 +643,14 @@ object StructType extends AbstractDataType {
         DecimalType.Fixed(rightPrecision, rightScale)) =>
         if ((leftPrecision == rightPrecision) && (leftScale == rightScale)) {
```

Comment:
how about this
```
if(leftScale == rightScale) {
  DecimalType(leftPrecision.max(rightPrecision), leftScale)
} else {
  throw QueryExecutionErrors.cannotMergeDecimalTypesWithIncompatibleScaleError
}
```

I think we don't need to have two different error messages. We just need to point out the root cause, which is the incompatible scale.

Reply:
Yes, makes sense. I updated the code.